regolith/file_system.go

```diff
@@ -869,3 +870,110 @@ func MoveOrCopy(
 	}
 	return nil
 }
+
+// SyncDirectories copies the source to destination while checking size and modification time.
+func SyncDirectories(
+	source string, destination string, makeReadOnly bool, copyParentAcl bool,
+) error {
+	// Make destination parent if not exists
+	destinationParent := filepath.Dir(destination)
+	if _, err := os.Stat(destinationParent); os.IsNotExist(err) {
+		err = os.MkdirAll(destinationParent, 0755)
+		if err != nil {
+			return burrito.WrapErrorf(
+				err, osMkdirError, destinationParent)
+		}
+	}
+	err := filepath.Walk(source, func(srcPath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		relPath, err := filepath.Rel(source, srcPath)
+		if err != nil {
+			return burrito.WrapErrorf(err, filepathRelError, source, srcPath)
+		}
+		destPath := filepath.Join(destination, relPath)
+
+		destInfo, err := os.Stat(destPath)
+		if err != nil && !os.IsNotExist(err) {
+			return burrito.WrapErrorf(err, osStatErrorAny, destPath)
+		}
+		if (err != nil && os.IsNotExist(err)) || info.ModTime() != destInfo.ModTime() || info.Size() != destInfo.Size() {
+			if info.IsDir() {
+				return os.MkdirAll(destPath, info.Mode())
+			}
+			Logger.Debugf("SYNC: Copying file %s to %s", srcPath, destPath)
+			return copyFile(srcPath, destPath, info)
+		} else {
+			Logger.Debugf("SYNC: Skipping file %s", srcPath)
+		}
+		return nil
+	})
+	if err != nil {
+		return burrito.WrapErrorf(err, osCopyError, source, destination)
+	}
+
+	// Remove files/folders in destination that are not in source
+	err = filepath.Walk(destination, func(destPath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		relPath, err := filepath.Rel(destination, destPath)
+		if err != nil {
+			return burrito.WrapErrorf(err, filepathRelError, destination, destPath)
+		}
+		srcPath := filepath.Join(source, relPath)
+		if _, err := os.Stat(srcPath); os.IsNotExist(err) {
+			Logger.Debugf("SYNC: Removing file %s", destPath)
+			return os.RemoveAll(destPath)
+		}
+		return nil
+	})
+
+	if err != nil {
+		return burrito.WrapErrorf(err, osRemoveError, source, destination)
+	}
+
+	//TODO: copy ACL. To be honest, I have no clue why it was there in the first place
```

Comment:
The ACL thing has been added so that Minecraft has permission to use the files. You can read about it in issue #85

Reply:
From what I understood in that issue, the bug happens only when moving files. Sync function always copies the files, which makes me wonder whether it's needed in this case. Also seems like it can be skipped if move function fails in original MoveOrCopy

Reply:
I'm not sure. The best way to check is to look into the directory properties shown in 85.

Reply:
I think you're right. There should be no issues when you copy:
https://learn.microsoft.com/en-us/troubleshoot/windows-server/windows-security/inherited-permissions-not-automatically-update